EssentialsDiscord/src/main/java/net/essentialsx/api/v2/services/discord/MessageType.java

```diff
@@ -0,0 +1,70 @@
+package net.essentialsx.api.v2.services.discord;
+
+/**
+ * Indicates the type of message being sent and its literal channel name used in the config.
+ */
+public final class MessageType {
+    private final String key;
+    private final boolean player;
+
+    /**
+     * Creates a {@link MessageType} which will send channels to the specified channel key.
+     * <p>
+     * The message type key may only contain: lowercase letters, numbers, and dashes.
```

Comment:
An invalid message type shouldn't be usable anywhere anyway - might be better to validate on construction rather than registration.